Add xpu device support. (#1)
oneapi-src/text-data-generation: README.md

**YAML file**                                 | **Environment Name** |  **Configuration** |
| :---: | :---: | :---: |
`env/intel/text-intel-torch-cpu.yml`             | `text-intel-torch` | Python=3.9.7, PyTorch v1.13, Intel® Extension for PyTorch v1.13, Intel® Neural Compressor v2.0.0 |
`env/intel/text-intel-torch-xpu.yml`             | `text-intel-torch` | Python=3.9.7, PyTorch v1.13, Intel® Extension for PyTorch v1.13+xpu, Intel® Neural Compressor v2.0.0 |

### Optimized Solution Setup

> Follow the below conda installation commands to setup the Intel® oneAPI optimized PyTorch environment for model training and text generation. 

Reviewer: ```suggestion
Follow the below conda installation commands to setup the Intel® oneAPI optimized PyTorch environment for model training and text generation. Please choose `env/intel/text-intel-torch-xpu.yml` if you have Intel GPU available.

```

```sh

> conda env create -f env/intel/text-intel-torch-cpu.yml # For CPU
> conda env create -f env/intel/text-intel-torch-xpu.yml # For XPU

Reviewer: ```suggestion
conda env create -f env/intel/text-intel-torch-cpu.yml # For CPU
or
conda env create -f env/intel/text-intel-torch-xpu.yml # For XPU
```

```

*Activate stock conda environment*
Use the following command to activate the environment that was created:
```sh
conda activate text-intel-torch
```

> Especially for XPU platform:

Reviewer: ```suggestion
Please perform this additional installation step only if you are using an Intel GPU, CPU users can skip this step:

```

ipexrun --use_logical_core --enable_tcmalloc src/finetune_model.py --model_config configs/config_base.yml --data_path data/abcnews-date-text.csv --save_path saved_models/gpt2-medium-finetuned-intel --intel
```

> **For intel xpu training**, 'xpu' device is needed to add to 'python/site-packages/transformers'.    
> It can be changed manually as the following:

Reviewer: ```suggestion
**For Intel GPU training**, device 'xpu' must be added to 'python/site-packages/transformers/training_args.py as shown in the below diff at lines'.    
It can be changed manually follows:

```

> ```python

Reviewer: Please update the diff with the latest one (one that also has "import intel_extension_for_pytorch as ipex")

diff --git a/training_args.py b/training_args.py
index 1a90710..9421ccd 100644
--- a/training_args.py
+++ b/training_args.py
@@ -1466,6 +1466,8 @@ class TrainingArguments:
                 torch.distributed.init_process_group(
                     backend=self.xpu_backend, rank=rank, world_size=size, timeout=self.ddp_timeout_delta
                 )
+        elif torch.xpu.is_available():
+            device = torch.device("xpu")
         elif is_torch_tpu_available():
             device = xm.xla_device()
             self._n_gpu = 0
```

> Or run this python code only once:

Reviewer: ```suggestion
To apply the patch automatically, please run the following Python script

```

```bash

> $ cd src
> $ python ./apply_xpu_patch.py

Reviewer: ```suggestion
$ cd src
$ python ./apply_xpu_patch.py

and then run the fine tuning step as follows:
```sh
ipexrun --use_logical_core --enable_tcmalloc src/finetune_model.py --model_config configs/config_base.yml --data_path data/abcnews-date-text.csv --save_path saved_models/gpt2-medium-finetuned-intel --intel
```
```
